spec/vagrant/core_spec.rb

```diff
@@ -0,0 +1,341 @@
+
+describe file('/etc/mistral/wf_trace_logging.conf') do
+  it { should be_file }
+  it { should contain 'args=("/var/log/mistral_wf_trace.log",)' }
+end
+
+describe file('/etc/init/mistral.conf'), :if => os[:family] == 'ubuntu' do
+  it { should be_file }
+  it { should contain MISTRAL_INIT_COMMAND }
+end
+
+describe file('/etc/systemd/system/mistral.service'), :if => os[:family] == 'redhat' do
+  it { should be_file }
+  it { should contain MISTRAL_INIT_COMMAND }
+end
+
+describe file('/etc/mistral/actions/st2mistral') do
+  it { should be_directory }
+end
+
+describe file('/etc/mistral/actions/st2mistral/.git') do
+  it { should be_directory }
+end
+
+describe port(8989) do
+  it { should be_listening }
+end
+
+# st2api
+describe port(9101) do
+  it { should be_listening }
+end
+
+# TODO: This needs an init script
+describe service('st2api') do
+  # it { should be_enabled }
+  # it { should be_running }
+end
+
+describe command('ps ax') do
+  its(:stdout) { should match(/st2api/) }
+end
+
+# st2auth
+describe port(9100) do
+  it { should be_listening }
+end
+
+# TODO: This needs an init script
+describe service('st2auth') do
+  # it { should be_enabled }
+  # it { should be_running }
+end
+
+describe command('ps ax') do
+  its(:stdout) { should match(/st2auth/) }
+end
+
+describe command('st2 auth testu -p testp') do
+  its(:exit_status) { should eq 0 }
+  its(:stdout) { should match(/token/) }
+end
+
+# actionrunner
+# TODO: This needs an init script
+describe service('st2actionrunner') do
+  # it { should be_enabled }
+  # it { should be_running }
+end
+
+describe command('ps ax') do
+  its(:stdout) { should match(/st2actionrunner/) }
+end
+
+# notifier
+# TODO: This needs an init script
+describe service('st2notifier') do
+  # it { should be_enabled }
+  # it { should be_running }
+end
+
+describe command('ps ax') do
+  its(:stdout) { should match(/st2notifier/) }
+end
+
+# rules engine
+# TODO: This needs an init script
+describe service('st2rulesengine') do
+  # it { should be_enabled }
+  # it { should be_running }
+end
+
+describe command('ps ax') do
+  its(:stdout) { should match(/st2rulesengine/) }
+end
+
+# sensor container
+# TODO: This needs an init script
+describe service('st2sensorcontainer') do
+  # it { should be_enabled }
+  # it { should be_running }
+end
+
+describe command('ps ax') do
+  its(:stdout) { should match(/st2sensorcontainer/) }
+end
+
+# results tracker
+# TODO: This needs an init script
+describe service('st2resultstracker') do
+  # it { should be_enabled }
+  # it { should be_running }
+end
+
+describe command('ps ax') do
+  its(:stdout) { should match(/st2resultstracker/) }
+end
+
+# st2web
+WEBUI_CONFIG_PATH='/opt/stackstorm/static/webui/config.js'
+AUTH_FILE_PATH='/etc/st2/htpasswd'
+
+describe file(WEBUI_CONFIG_PATH) do
+  it { should be_file }
+  it { should contain "name: 'StackStorm'" }
+  it { should contain "url: '//:9101'" }
+  it { should contain "auth: '//:9100'" }
+end
+
+describe port(8080) do
```

Comment:
These are purposely split like this, because I intend to have them in their own files after a refactor round. That'll be necessary with docker testing. So, this makes it easy to copy/paste, which literally is next.